UndertaleModLib/UndertaleLists.cs

```diff
@@ -7,171 +7,214 @@
+            // Set capacity directly
+            internalList.Capacity = capacity;
+        }
 
+        /// <summary>
+        /// Sets the capacity of the internal list of this collection.
+        /// </summary>
         public void SetCapacity(int capacity)
         {
-            try
-            {
-                internalList.Capacity = capacity;
-            }
-            catch (Exception e)
-            {
-                throw new UndertaleSerializationException($"{e.Message}\nwhile trying to \"SetCapacity()\" of \"UndertalePointerList<{typeof(T).FullName}>\".");
-            }
+            internalList.Capacity = capacity;
         }
+
+        /// <summary>
+        /// Sets the capacity of the internal list of this collection.
+        /// </summary>
         public void SetCapacity(uint capacity) => SetCapacity((int)capacity);
 
+        /// <summary>
+        /// Adds an item to the internal list of this collection (that is, without notifying any observers).
+        /// </summary>
+        /// <param name="item">Item to add.</param>
         public void InternalAdd(T item)
         {
             internalList.Add(item);
         }
     }
 
-    public class UndertaleSimpleList<T> : UndertaleListBase<T> where T : UndertaleObject, new()
+    /// <summary>
+    /// Simple list object, serialized as a 32-bit count followed immediately by all objects in the list.
+    /// </summary>
+    public class UndertaleSimpleList<T> : UndertaleObservableList<T>, UndertaleObject where T : UndertaleObject, new()
     {
         /// <inheritdoc />
-        public override void Serialize(UndertaleWriter writer)
+        public void Serialize(UndertaleWriter writer)
         {
-            writer.Write((uint)Count);
-            for (int i = 0; i < Count; i++)
+            // Write list count
+            int count = Count;
+            writer.Write((uint)count);
+
+            // Write actual objects
+            int i = 0;
+            try
             {
-                try
-                {
-                    writer.WriteUndertaleObject<T>(this[i]);
-                }
-                catch (UndertaleSerializationException e)
+                for (; i < count; i++)
                 {
-                    throw new UndertaleSerializationException(e.Message + "\nwhile writing item " + (i + 1) + " of " + Count + " in a list of " + typeof(T).FullName, e);
+                    writer.WriteUndertaleObject(this[i]);
                 }
             }
+            catch (UndertaleSerializationException e)
+            {
+                throw new UndertaleSerializationException($"{e.Message}\nwhile writing item {i + 1} of {count} in a list of {typeof(T).FullName}", e);
+            }
         }
 
         /// <inheritdoc />
-        public override void Unserialize(UndertaleReader reader)
+        public void Unserialize(UndertaleReader reader)
         {
+            // Read count and initialize list with that capacity
```

Comment:
wording nit, but i'd prefer something like "set list with that capacity" instead.